toolchain/mingw-toolchain.xml

```diff
@@ -45,14 +47,17 @@
 </linker>
 
 <linker id="exe" exe="g++">
-  <exe name="g++.exe"/>
+  <exe name="g++.exe" unless="linux_host"/>
+  <exe name="i686-w64-mingw32-g++" if="linux_host"/>
   <flag value="-debug" if="debug"/>
   <flag value="-Wl,--enable-auto-import"/>
   <flag value="-mwindows" if="SUBSYSTEMWINDOWS" />
   <flag value="-m32" unless="HXCPP_M64"/>
   <flag value="-static" if="no_shared_libs"/>
-  <flag value="-static-libgcc" if="no_shared_libs"/>
-  <flag value="-static-libstdc++" if="no_shared_libs"/>
+  <flag value="-static-libgcc" if="no_shared_libs" unless="linux_host"/>

```

Comment:
unless "linux_host" is indicated, any other host platform can compile as if the code was not modified.